File: solutions/python/question1.py
"""Q1 — Rede de armários de encomendas.

Abra questions/q1_parcel_network/LEVEL_1.md e preserve as assinaturas públicas.
"""


class ParcelLockerService:
    def add_locker(self, locker_id: str, capacity: int) -> bool:
        return False

    def store_parcel(
        self, timestamp: int, parcel_id: str, locker_id: str, size: int
    ) -> bool:
        return False

    def collect_parcel(self, timestamp: int, parcel_id: str) -> bool:
        return False

    def get_load(self, timestamp: int, locker_id: str) -> int | None:
        return None

    def move_parcel(
        self, timestamp: int, parcel_id: str, target_locker_id: str
    ) -> bool:
        return False

    def top_lockers(self, timestamp: int, n: int) -> list[str]:
        return []

    def schedule_move(
        self,
        timestamp: int,
        parcel_id: str,
        target_locker_id: str,
        execute_at: int,
    ) -> str | None:
        return None

    def cancel_scheduled_move(self, timestamp: int, move_id: str) -> bool:
        return False

    def process_scheduled(self, timestamp: int) -> list[str]:
        return []

    def merge_lockers(
        self, timestamp: int, source_locker_id: str, target_locker_id: str
    ) -> bool:
        return False

    def get_parcel_history(self, timestamp: int, parcel_id: str) -> list[str]:
        return []

File: solutions/typescript/question1.ts
/**
 * Q1 — Rede de armários de encomendas.
 *
 * Abra questions/q1_parcel_network/LEVEL_1.md e implemente somente o nível
 * atualmente desbloqueado. Preserve todas as assinaturas públicas.
 */
export class ParcelLockerService {
  addLocker(lockerId: string, capacity: number): boolean {
    return false;
  }

  storeParcel(timestamp: number, parcelId: string, lockerId: string, size: number): boolean {
    return false;
  }

  collectParcel(timestamp: number, parcelId: string): boolean {
    return false;
  }

  getLoad(timestamp: number, lockerId: string): number | null {
    return null;
  }

  moveParcel(timestamp: number, parcelId: string, targetLockerId: string): boolean {
    return false;
  }

  topLockers(timestamp: number, n: number): string[] {
    return [];
  }

  scheduleMove(
    timestamp: number,
    parcelId: string,
    targetLockerId: string,
    executeAt: number
  ): string | null {
    return null;
  }

  cancelScheduledMove(timestamp: number, moveId: string): boolean {
    return false;
  }

  processScheduled(timestamp: number): string[] {
    return [];
  }

  mergeLockers(timestamp: number, sourceLockerId: string, targetLockerId: string): boolean {
    return false;
  }

  getParcelHistory(timestamp: number, parcelId: string): string[] {
    return [];
  }
}
